auth/server/src/main/java/com/seagates3/policy/IAMPolicyAuthorizer.java

```diff
@@ -20,17 +20,151 @@
 
 package com.seagates3.policy;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
+import org.json.JSONException;
+import org.json.JSONObject;
+import com.seagates3.model.User;
+import com.amazonaws.auth.policy.Action;
+import com.amazonaws.auth.policy.Statement;
+import com.amazonaws.auth.policy.Statement.Effect;
+import com.seagates3.exception.DataAccessException;
 import com.seagates3.model.Requestor;
 import com.seagates3.response.ServerResponse;
+import com.seagates3.response.generator.AuthorizationResponseGenerator;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
 
 public
 class IAMPolicyAuthorizer extends PolicyAuthorizer {
 
   @Override public ServerResponse authorizePolicy(
       Requestor requestor, Map<String, String> requestBody) {
-    // TODO Auto-generated method stub
-    return null;
+
+    ServerResponse serverResponse = null;
+
+    // authorizePolicy will return NULL if no relevant entry found in policy
+    // authorized if match is found
+    // AccessDenied if Deny found
+    String requestedOperation = null;
+    try {
+      if (requestBody.get("Action") != null &&
+          !requestBody.get("Action").contains("Authorize")) {
+        requestedOperation = "iam:" + requestBody.get("Action");
+      } else {
+        requestedOperation =
+            identifyOperationToAuthorize(requestBody.get("S3Action"));
+      }
+      LOGGER.debug("operation to authorize - " + requestedOperation);
+      if (requestedOperation != null && requestor != null &&
+          requestor.getPolicyDocuments() != null) {
+        for (String policyToAuthorize : requestor.getPolicyDocuments()) {
+          // authorize policy
+          serverResponse =
+              authorizeOperation(requestBody, requestedOperation.toLowerCase(),
+                                 policyToAuthorize, requestor.getUser());
+          if (serverResponse != null &&
+              serverResponse.getResponseStatus() != HttpResponseStatus.OK) {
+            break;
+          }
+        }
+      }
+    }
+    catch (Exception e) {
+      LOGGER.error("Exception while authorizing", e);
```

Comment:
to make it more readable
"Exception while authorizing -"

Reply:
fixed